### PR TITLE
2018 edition, upgrade dependencies, remove `nightly` feature

### PR DESCRIPTION
**Pull request by akhilles**
`spin::Mutex` seems more appropriate for this use case. It can eventually support `#![no_std]` as well. I can revert back to `lazy_static` + `std::sync::Mutex` if needed.

### CODE REVIEW

**Comment by akhilles** [comment]
This change bumps MSRV to 1.31.0. So, minor release version was incremented.

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +28,10 @@ addons:
 before_install:
 - source ./ci/before_install.sh
 
+matrix:
+  fast_finish: true
+  allow_failures:
+  - rust: nightly
```
Comment (fitzgen) [inline .travis.yml]: Why was this necessary? I'd prefer not to allow failures unless there is a reason it is flaky right now.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ rust:
 - nightly
 - beta
 - stable
+cache: cargo
```
Comment (fitzgen) [inline .travis.yml]: Can you also add testing on rust 1.31 above this?

**Comment by akhilles** [comment]
Should `nightly` feature be removed as part of this change?

File: Cargo.toml [hunk 3]
```diff
@@ -16,8 +17,11 @@ repository = "gimli-rs/findshlibs"
...
 
 [dependencies]
-lazy_static = "1.0.0"
-libc = "0.2.55"
+libc = "0.2.65"
 
+[target.'cfg(target_os = "macos")'.dependencies]
+lazy_static = "1.4"
+
+# TODO: remove?
```
Comment (fitzgen) [inline Cargo.toml]: Yeah let's remove it if it isn't used anymore.

**Pull request merged**